packages/diagrams/frontend/sirius-components-diagrams/src/renderer/tools/useSingleClickTool.tsx

```diff
@@ -67,25 +68,31 @@ export const useSingleClickTool = (): UseSingleClickToolValue => {
   const { showDialog } = useDialog();
   const { showImpactAnalysisDialog } = useImpactAnalysisDialog();
 
-  const [invokeSingleClickOnDiagramElementTool, { loading, data }] = useMutation<
+  const [invokeSingleClickOnDiagramElementTool, { loading, data, error }] = useMutation<
     GQLInvokeSingleClickOnDiagramElementToolData,
     GQLInvokeSingleClickOnDiagramElementToolVariables
-  >(invokeSingleClickOnDiagramElementToolMutation, {
-    onCompleted(data) {
-      const { invokeSingleClickOnDiagramElementTool } = data;
-      if (isInvokeSingleClickSuccessPayload(invokeSingleClickOnDiagramElementTool)) {
-        const { newSelection } = invokeSingleClickOnDiagramElementTool;
-        if (newSelection?.entries.length ?? 0 > 0) {
-          setSelection(newSelection);
+  >(invokeSingleClickOnDiagramElementToolMutation);
+
+  useEffect(() => {
```

Comment:
Now that we are sure the mutation is not computed on an unmounted component we can handle data/error the usual way